src/node_debug_options.h

```diff
@@ -0,0 +1,52 @@
+#ifndef SRC_NODE_DEBUG_OPTIONS_H_
+#define SRC_NODE_DEBUG_OPTIONS_H_
+
+#include <string>
+
+// Forward declaration to break recursive dependency chain with src/env.h.
+namespace node {
+
+enum class DebugAgentType {
+  kNone,
+  kDebugger,
+#if HAVE_INSPECTOR
+  kInspector
+#endif  // HAVE_INSPECTOR
+};
+
+class DebugOptions {
+ public:
+  DebugOptions();
+  bool ParseOption(const std::string& option);
+  DebugAgentType EnabledAgentType() const {
+#if HAVE_INSPECTOR
+    if (inspector_enabled_) {
+      return DebugAgentType::kInspector;
+    }
+#endif  // HAVE_INSPECTOR
+    if (debug_enabled_) {
+      return DebugAgentType::kDebugger;
+    }
+    return DebugAgentType::kNone;
+  }
+  void EnableDebugAgent(DebugAgentType type);
+  bool ToolsServerEnabled();
+  bool WaitForConnect() const { return wait_connect_; }
+  std::string HostName() const { return host_name_; }
+  int Port() const;
+  void SetPort(int port) { port_ = port; }
```

Comment:
It's not a big thing but I would call these `hostname()`, `port()` and `set_port()`.  The sort-of convention in core is that CamelCase methods imply computation or side effects whereas snake_case methods are simple getters and setters.